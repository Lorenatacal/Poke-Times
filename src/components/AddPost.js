import React, {Component} from 'react'
import { connect } from 'react-redux';
import { Link, NavLink, withRouter } from 'react-router-dom'
import Form from './Form'

class AddPost extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            showComponent: false,
        };
        this.onClick = this.onClick.bind(this);
    }

   onClick() {
       this.setState({
           showComponent: true,
       });
   }

    render() {
        return (
            <div className="center">
                <button className="btn grey" onClick={this.onClick}>
                  Add Post
                </button>
                { this.state.showComponent && <Form /> }
            </div>
        );
    }
}

export default AddPost