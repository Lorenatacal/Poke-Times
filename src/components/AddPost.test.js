import React, { Component } from 'react';
import AddPost from './AddPost';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

test("AddPost should render correctly", () => {
    const wrapper = Enzyme.shallow(<AddPost />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

test("AddPost should render correctly onClick", () => {
    const wrapper = Enzyme.shallow(<AddPost />);
    const AddPostButton = wrapper.find('[data-name="AddPostButton"]');
    AddPostButton.simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
})