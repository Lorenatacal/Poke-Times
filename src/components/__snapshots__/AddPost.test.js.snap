// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPost should render correctly 1`] = `
<div
  className="center"
>
  <button
    className="btn grey"
    data-name="AddPostButton"
    onClick={[Function]}
  >
    Add Post
  </button>
</div>
`;

exports[`AddPost should render correctly onClick 1`] = `
<div
  className="center"
>
  <button
    className="btn grey"
    data-name="AddPostButton"
    onClick={[Function]}
  >
    Add Post
  </button>
  <Connect(AddForm) />
</div>
`;
